core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -279,70 +282,78 @@ public void removeInaccessibleStorage() {
   }
 
   @Override
-  public void requestSpace(long sessionId, long blockId, long additionalBytes)
-      throws WorkerOutOfSpaceException, IOException {
+  public void requestSpace(long sessionId, long blockId, long additionalBytes) {
     mLocalBlockStore.requestSpace(sessionId, blockId, additionalBytes);
   }
 
   @Override
   public List<BlockStatus> load(List<Block> blocks, String tag, OptionalLong bandwidth) {
     ArrayList<CompletableFuture<BlockStatus>> futures = new ArrayList<>();
+    ArrayList<BlockStatus> errors = new ArrayList<>();
+    long sessionId = IdUtils.createSessionId();
     for (Block block : blocks) {
-      CompletableFuture<Void> future = CompletableFuture.runAsync(() -> {
+      long blockId = block.getBlockId();
+      long blockSize = block.getBlockSize();
+      BlockWriter blockWriter;
+      UfsIOManager manager;
+      BlockStoreLocation loc =
+          BlockStoreLocation.anyDirInTier(WORKER_STORAGE_TIER_ASSOC.getAlias(0));
+      try {
+        manager = mUnderFileSystemBlockStore.getOrAddUfsIOManager(block.getMountId());
+        if (bandwidth.isPresent()) {
+          manager.setQuota(tag, bandwidth.getAsLong());
+        }
+        requestSpace(sessionId, blockId, blockSize);
+        mLocalBlockStore.createBlock(sessionId, blockId, AllocateOptions.forCreate(blockSize, loc));
+        blockWriter = mLocalBlockStore.createBlockWriter(sessionId, blockId);
+      } catch (IOException | AlluxioRuntimeException e) {
+        AlluxioRuntimeException exception = AlluxioRuntimeException.from(e);
+        errors.add(BlockStatus.newBuilder().setBlock(block)
+            .setCode(exception.getStatus().getCode().value()).setMessage(e.getMessage())
+            .setRetryable(exception.isRetryable()).build());
+        continue;
+      }
+      CompletableFuture<Void> future = CompletableFuture.completedFuture(null);
```

Comment:
This is kinda weird. I know it's convenient to write this way but i'd still prefer to not have an unnecessary future.